sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -836,11 +836,81 @@ class PageRange(DictMixin):
         Start of page range in bytes.
     :param int end:
         End of page range in bytes.
+    :ivar bool cleared:
+        Whether the range has been cleared.
     """
 
-    def __init__(self, start=None, end=None):
+    def __init__(self, start=None, end=None, *, cleared=False):
         self.start = start
         self.end = end
+        self.cleared = cleared
+
+    @staticmethod
+    def _parse_page_list(page_list):
+        """Parse a generated PageList into a single list of PageRange sorted by start.
+        """
+        page_ranges = page_list.page_range
+        clear_ranges = page_list.clear_range
+
+        ranges = []
+        p_i, c_i = 0, 0
+
+        # Combine page ranges and clear ranges into single list, sorted by start
```

Comment:
Is this the logic that the .NET lib used to sort?
Does this hold true with the next page - i.e. the pages will be ordered according to start?

Reply:
Yes, I based this off of .NET. Looking at the service documentation, the list is guaranteed to come back ordered by range, so I assume the page and clear lists are already sorted.

https://docs.microsoft.com/en-us/rest/api/storageservices/get-page-ranges#response-body
"The response body includes a list of non-overlapping valid page ranges, sorted by increasing address page range."